### PR TITLE
Test ChassImporter

### PR DESCRIPTION
**Pull request by freeatnet**
This PR adds some tests for `ChassImporter`. Aside from testing `ChassImporter`, the goal of this PR is to verify our testing setup works and to give a small example of what RSpec tests look like. The tests are located in spec/services/chass_importer_spec.rb.

You can run tests once like so:
```
docker-compose run -eRAILS_ENV=test rails-app rake spec
```

…or like so when doing development (automatically runs applicable tests whenever files change):
```
docker-compose run -eRAILS_ENV=test rails-app guard
```

❗️ Tests in this PR will fail. Fixes are proposed in uoft-tapp/tapp@85085956c58b63812d9ab40e42b458b0fe6e26eb

👂 Please let me know if the testing setup works for you. Also please let me know if you have any questions/ideas/suggestions on the testing front.

### CODE REVIEW

**Review by simpslandyy — APPROVED** [review]
I ran the tests, they ran fine on my setup :)
Just to verify, I got : 
``` 23 examples, 4 failures, 8 pending ```

& the failures were:
``` 
when used on an empty file raises a descriptive error (FAILED - 1)

when parsing courses:
    with no round_id raises a descriptive error (FAILED - 2)
    from a file missing courses key raises a descriptive error (FAILED - 3)
    with more than one round_id raises a descriptive error (FAILED - 4)
```

**Comment by freeatnet** [comment]
@simpslandyy That's the expected failures 👍

**Comment by michellemtchai** [comment]
Are these the expected results?
````
Finished in 0.21317 seconds (files took 5.37 seconds to load)
23 examples, 4 failures, 8 pending

Failed examples:

rspec ./spec/services/chass_importer_spec.rb:13 # ChassImporter when used on an empty file raises a descriptive error
rspec ./spec/services/chass_importer_spec.rb:28 # ChassImporter when parsing courses from a file missing courses key raises a descriptive error
rspec ./spec/services/chass_importer_spec.rb:35 # ChassImporter when parsing courses with no round_id raises a descriptive error
rspec ./spec/services/chass_importer_spec.rb:42 # ChassImporter when parsing courses with more than one round_id raises a descriptive error
````

**Comment by michellemtchai** [comment]
Okay. This is the result after running `git checkout 23-data-importer-tests; git fetch; git reset --hard origin/23-data-importer-tests`:
````
Finished in 0.2204 seconds (files took 1.67 seconds to load)
23 examples, 0 failures, 8 pending
````

File: spec/services/chass_importer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+      let (:mock_json) {'{"courses" : []}'}
+      it "raises a descriptive error" do
+        expect { subject }.to raise_error(KeyError, /key not found: "applicants"/)
+      end
+    end
+
+    context "from a file with no applicants" do
+      let (:mock_json) {'{"courses": [], applicants": []}'}
+      it "does not raise any errors" do
+        "No reason given"
```
Comment (freeatnet) [inline spec/services/chass_importer_spec.rb]: Hmm?

File: spec/services/chass_importer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+      end
+    end
+
+    context "from a file with a non-UtorID applicant" do
+      let (:mock_json) { File.read("./spec/support/chass_data/no_utorid_applicant.json") }
+      it "raises ActiveRecord::NotNullViolation exception" do
+        expect { subject }.to raise_error(ActiveRecord::NotNullViolation,
+          /PG::NotNullViolation: ERROR:  null value in column "utorid" violate/)
```
Comment (freeatnet) [inline spec/services/chass_importer_spec.rb:158]: 💯

**Comment by freeatnet** [comment]
@michellemtchai I get a single failing test right now:
```
  1) ChassImporter when importing applications from a file with non-existent courses positions raise a descriptive error for courses
     Failure/Error: expect { subject }.to raise_error(StandardError, /no such course positions/)
       expected StandardError with message matching /no such course positions/ but nothing was raised
     # ./spec/services/chass_importer_spec.rb:207:in `block (4 levels) in <top (required)>'
```

Did you make a change somewhere else?

Other than that, looks good!

**Comment by michellemtchai** [comment]
The single failing test is the expected result.

File: spec/services/chass_importer_spec.rb (hunk 1 above)
Comment (michellemtchai) [reply spec/services/chass_importer_spec.rb]: I had trouble doing this test. What is usually done in this sort of "no error" tests?

Comment (freeatnet) [reply spec/services/chass_importer_spec.rb]: Having the importer run and `expect` it `.to_not raise_error` is, I think, a reasonable outcome

Comment (michellemtchai) [reply spec/services/chass_importer_spec.rb]: Okay. I'll add that.

**Pull request merged**